packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -1450,6 +1460,64 @@ export class AiAgentService {
             return filteredExplore;
         };
 
+        const findFields: FindFieldFn = async ({ fieldSearchQuery }) => {
+            const userAttributes =
+                await this.userAttributesModel.getAttributeValuesForOrgMember({
+                    organizationUuid,
+                    userUuid: user.userUuid,
+                });
+
+            const agentSettings = await this.getAgentSettings(user, prompt);
+
+            const catalogItems = await this.catalogService.searchCatalog({
+                projectUuid,
+                catalogSearch: {
+                    type: CatalogType.Field,
+                    searchQuery: fieldSearchQuery.name,
+                },
+                context: CatalogSearchContext.AI_AGENT,
+                // TODO: make this paginated
+                paginateArgs: {
+                    page: 1,
+                    pageSize: 5,
+                },
+                userAttributes,
+                yamlTags: agentSettings.tags,
+            });
+
+            // TODO: we should not filter here, we should return all the fields
```

Comment:
Sorry I don't understand the comment - from what I understand the filter is acting like a type guard right? because `searchCatalog` is already filtering by type **field**

Reply:
correct!